src/basic/terminal-util.c

```diff
@@ -1135,14 +1135,19 @@ int open_terminal_in_namespace(pid_t pid, const char *name, int mode) {
 }
 
 bool colors_enabled(void) {
-        const char *colors;
+        static int enabled = -1;
 
-        colors = getenv("SYSTEMD_COLORS");
-        if (!colors) {
-                if (streq_ptr(getenv("TERM"), "dumb"))
-                        return false;
-                return on_tty();
+        if (_unlikely_(enabled < 0)) {
```

Comment:
Is _unlikely_() really appropriate here? I figure quite often this might just be called a single time, in which case you add a penalty here... If it isn't clear what is right, it's better not to use _likely_ or _unlikely_